plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/TestBigQueryConnectorTest.java

```diff
@@ -621,6 +622,35 @@ public void testBigQuerySnapshotTable()
         }
     }
 
+    @Test
+    public void testQueryCache()
+    {
+        Session queryCacheSession = Session.builder(getSession())
+                .setCatalogSessionProperty("bigquery", "query_cache_enabled", "true")
+                .build();
+        Session createNeverDisposition = Session.builder(getSession())
+                .setCatalogSessionProperty("bigquery", "query_cache_enabled", "true")
```

Comment:
Q: Would it make sense to use the constants defined for the settings in the test?
```

import static io.trino.plugin.bigquery.BigQuerySessionProperties.QUERY_CACHE_ENABLED;
import static io.trino.plugin.bigquery.BigQuerySessionProperties.CREATE_DISPOSITION_TYPE;
[...]
.setCatalogSessionProperty("bigquery", QUERY_CACHE_ENABLED "true")

```

Reply:
Personally, I don't agree with the justification "If we [...] change the session property name accidentally" - 
If you want to test that the constant doesn't change, I would create an explicit test:
```
@Test
public void testConstantValues()
{
    assertEquals(QUERY_CACHE_ENABLED, "query_cache_enabled");
}
```

Here are examples of session properties being used as constants (and the value is a constant as well)
```
            Session convertToVarchar = Session.builder(session)
                    .setCatalogSessionProperty("postgresql", UNSUPPORTED_TYPE_HANDLING, CONVERT_TO_VARCHAR.name())
                    .build();
```

Why are the constants defined as `public static final` if they are used nowhere outside of BigQuerySessionProperties.java ? Change them to `private static` unless you wish to use them as constants in tests.

Reply:
We should fix the other occurrence then. Good point about making them private if not needed. Let's check that and change accordingly.

Reply:
The other occurence is a usage but note that TestTypeHandlingJdbcConfig already covers it by using explicit string and not the constant.

Reply:
I think it's strange to define constants and not use them (without a compelling reason not to).

Comparing the (current) 2 approaches:
```
                // Explicitly disable join pushdown. The purpose of the this test is to verify partial TopN gets pushed down when Join is not.
                Session.builder(getSession())
                        .setCatalogSessionProperty(getSession().getCatalog().orElseThrow(), JOIN_PUSHDOWN_ENABLED, "false")
                        .build(),
```

versus:

```
        return Session.builder(session)
                .setCatalogSessionProperty(session.getCatalog().orElseThrow(), "join_pushdown_enabled", "true")
                .build();
```

Former allows me to search for usages of `SystemSessionProperties#JOIN_PUSHED_ENABLED` + find all code which sets / inspects this value.
The latter usage means I have to do a full text search for "join_pushdown_enabled" in order to find all usages.
It's worse for "common" property names like "cache_enabled" - is that `SystemSessionProperties#CACHE_ENABLED` or `BigQueryMetadata#CACHE_ENABLED` ? You would have to inspect each usage (clashing property names for the same context is obviously a bad example / would be bad for other reasons), but overall not using constants makes tracking usage more difficult and error prone. So I would write a test that the constants match the expected values, and then exclusively use constants in code.

(the consistency in the existing codebase (which is more hard-coding than constants, but not 100% either way) means we would need to consider updating the guidelines / existing usages + try to keep it consistent (if anyone else besides me see value in this approach)).

Reply:
@leveyja Feel free to continue on Slack `#dev` channel.